RecoHGCal/TICL/src/TracksterMomentumPluginBase.cc

```diff
@@ -0,0 +1,4 @@
+#include "RecoHGCal/TICL/interface/TracksterMomentumPluginBase.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+EDM_REGISTER_PLUGINFACTORY(TracksterMomentumPluginFactory, "TracksterMomentumPluginFactory");
```

Comment:
this is wrong, we shoud never register a plugin from a shared library

Reply:
@smuzaffar this is registering a factory, not a plugin. Doesn't this need to be in a shared library to be accessible?

Reply:
ah did not realize that @kpedro88 . Looking at cmssw , yes plugin factories are ok to be in shared libs.